app_dart/lib/src/model/appengine/task.dart

```diff
@@ -339,6 +408,27 @@ class Task extends Model<int> {
     _setStatusFromLuciStatus(build);
   }
 
+  /// Updates [Task] based on a Buildbucket [Build].
+  void updateFromBuildbucketBuild(bb.Build build) {
+    buildNumber = build.number!;
+
+    if (buildNumberList == null) {
+      buildNumberList = '$buildNumber';
+    } else {
+      final Set<String> buildNumberSet = buildNumberList!.split(',').toSet();
+      buildNumberSet.add(buildNumber.toString());
+      buildNumberList = buildNumberSet.join(',');
+    }
+
+    createTimestamp = build.startTime?.millisecondsSinceEpoch ?? 0;
+    startTimestamp = build.startTime?.millisecondsSinceEpoch ?? 0;
+    endTimestamp = build.endTime?.millisecondsSinceEpoch ?? 0;
+
+    attempts = (attempts ?? 0) + 1;
```

Comment:
Is each update here a rerun? We do not want to bump the `attempts` if this is just a status change, say `scheduled` -> `started`.

Usually we increase the `attempts` only when schedule/rerun a target.